replace consumer.owner ref with consumer.ownerId

vritant opened the pull request.

Ceiu commented on server/src/main/java/org/candlepin/pinsetter/tasks/ExportJob.java:
```diff
@@ -92,7 +92,7 @@ public void toExecute(JobExecutionContext context) throws JobExecutionException
         String apiUrl, Map<String, String> extensionData) {
         JobDataMap map = new JobDataMap();
-        map.put(JobStatus.OWNER_ID, consumer.getOwner().getKey());
+        map.put(JobStatus.OWNER_ID, consumer.getOwnerId());
```
This is a much needed sanity change, but without more changes in this class, this will break things. Gotta change the usage of the field to actual owner ID.

vritant commented:
@candlepin-pull-request-bot  test this please

vritant commented:
> In general this looks pretty good. One thing I'd like to see, however, is an object-specific lookup method which also performs state checking (ala getConsumerType). We use find quite often throughout the changes here, but never validate the input or output.

added a new method.

> Also, I wasn't able to see a clear distinction between when we used a curator or when we pass in a distinct owner with the consumer. Can you clarify the decision making process here?

basically, if a method stack already looked up an owner, I changed the method signature. there are exceptions though im sure. sometimes method is called in too many places, so simpler to use a curator.
Also, it would be overkill to inject curators everywhere we need owner id imo. I decided to decide on case by case basis. imo its not wise to try to enforce a thumb rule here.

> Comments for areas Github won't let me comment on:

    HealEntireOrgJob.java: We still use the owner key rather than the actual owner ID throughout.

I actually undid all the changes in the tasks that replaced owner key with id. going to put in a seperate card for it.

vritant commented on server/src/main/java/org/candlepin/model/OwnerCurator.java:
```diff
@@ -47,6 +48,36 @@ public OwnerCurator() {
+     * @return
+     *  An Owner instance for the specified ownerId
+     */
+    public Owner findOwnerById(String ownerId) {
```
@Ceiu  is this a suggestion or a more stricter request for change? I prefer using the owner id instead of another object cause that way its generic and can be used in more use cases irrespective of whether the owner id comes from a consumer or an owned object.

vritant replied:
why does it have to come from a persisted object? why should the method implementation care about the context in which the owner is being looked up? I dont see the harm in having a method that looks up an owner with owner id.

Ceiu replied:
Exactly for the reason you just stated. If we're passing in just a string, there should be no assumption of its source and, thus, no expectation of a given state. With such a method, doing state management is offloading work the caller should be doing. 

When we pass in an object from which we pull the ID, the method can now make assumptions about state of the ID value: non-null and valid. At this point, it's fair to be doing state management.

And, again, we already have ID-based lookups for owner ID.

vritant replied:
are you refering to find / get?

Ceiu replied:
Yes; especially if/when we get around to fixing our convention to match up with Hibernate's use of find/get.

vritant replied:
okay. I first implemented this PR while using find, but as you pointed out there is no validation, so I added this method which is basically find+validation.

Ceiu replied:
Right; the intent there was to call out that we need to either (a) add another method like the getConsumerType that validates the object's state while doing the lookup, or (b) add validation to the output of the calls to .find.

vritant replied:
im okay with another method like find with added validation to it, imo its better than (b) , and I dont quite like (a) cause that would mean one method for each object type.
is this is a suggestion or a more stricter request for change?

kahowell replied:
>  dont see the harm in having a method that looks up an owner with owner id.

+1

I don't buy arguments about the method making assumptions about state, I'd rather have employee defensive programming practices and do a check.

Let's stick with findOwnerById(String) as written.

> In the next scrum and we'll discuss it with the team.

I'm happy to discuss and revisit with convincing arguments :smile:, for now, what is here is fine. (We can always revisit outside the scope of this PR).

Ceiu replied:
You and I both know this will never be changed, and quite frankly, I'm sick of our conventions changing between curators and objects without any logic because "I felt like doing it this way"

Maintaining CP is a bear already, and every time we decide to do something differently on a whim is just raising that bar and I'm growing quite tired of seeing patchwork things go through without any value to justify the complete deviation from other methods. But whatever, what's another such patch when we already have a plethora of others, I suppose. :/
